pkg/proxy/ipvs/proxier.go

```diff
@@ -1772,11 +1772,6 @@ func (proxier *Proxier) cleanLegacyService(activeServices map[string]bool, curre
 		if _, ok := activeServices[cs]; !ok {
 			rsList, _ := proxier.ipvs.GetRealServers(svc)
 
-			// If we still have real servers graceful termination is not done
```

Comment:
In this case we would still try to delete a service even if it has real servers. This check should at least happen after the graceful termination checks below on line 1791. What do you think?

Reply:
if this check moved to line 1791, deleted service ip will still bind on the kube-ipvs0 link device even rs(pod) deleted later

Reply:
Oh, because on the next resync `currentServices` won't have the service anymore?

Reply:
In that case, let's just remove all the graceful termination logic for the delete case?

Reply:
Nevermind, `currentServices` should still have the deleted service IP cause of it queries via netlink. In what scenario would we stilll bind on kube-ipvs0 even if there are no real servers?

Reply:
Actually currentServices get from ipvs virtualService list rather than query netlink. In my opinion, ip address unbind along with ipvs rules deleted. When the same service recreate, new service ip will be bound on netlink.
In this issue we can see the deleted service IP all bound on netlink.[#80407](https://github.com/kubernetes/kubernetes/issues/80407)

Reply:
If I remember correctly, `GetVirtualServers` gets the virtual servers from netlink though?

Reply:
`If I remember correctly, GetVirtualServers gets the virtual servers from netlink though?`